examples_tf2_py/examples_tf2_py/waits_for_transform.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+
+import rclpy
+from rclpy.node import Node
+import rclpy.time
+from tf2_ros import LookupException
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+
+
+class WaitsForTransform(Node):
+    """
+    Wait for a transform using callbacks.
+
+    This class is an example of waiting for transforms.
+    It avoids blocking the executor by registering a callback to be called when a transform becomes
+    available.
+    """
+
+    def __init__(self):
+        super().__init__('example_waits_for_transform')
+
+        self._tf_buffer = Buffer()
+        self._tf_listener = TransformListener(self._tf_buffer, self)
+
+        self._output_timer = self.create_timer(1.0, self.on_timer)
+
+        self._lock = threading.Lock()
+        self._tf_future = None
+        self._from_frame = None
+        self._to_frame = None
+        self._when = None
```

Comment:
Hm, I think there's a race condition here at startup time.  I think it could be the case that `on_timer` fires *before* we initialize the lock and set these variables to None, in which case bad things may happen.  I'll suggest moving this initialization near the top of the constructor.

Reply:
I think it's ok. The timer callback can't be called until the node is added to an executor, and that can't happen until after `__init__()` is finished.

Reply:
Oh, you are totally right.  Sorry about that, will approve.